narwhals/_arrow/dataframe.py

```diff
@@ -288,8 +288,12 @@ def join(
     def drop(self, *columns: str) -> Self:
         return self._from_native_dataframe(self._native_dataframe.drop(list(columns)))
 
-    def drop_nulls(self) -> Self:
-        return self._from_native_dataframe(self._native_dataframe.drop_null())
+    def drop_nulls(self: Self, subset: str | list[str] | None) -> Self:
+        if subset is None:
+            return self._from_native_dataframe(self._native_dataframe.drop_null())
+        subset = [subset] if isinstance(subset, str) else subset
+        plx = self.__narwhals_namespace__()
+        return self.filter(~plx.any_horizontal(plx.col(*subset).is_null()))
```

Comment:
It's a bit of repeated code but avoids having any logic in `BaseFrame`. There are these situations in which polars works out of the box and for the rest implementing the solution using polars-like methods is just very neat